### PR TITLE
Etcd config source support TLS

### PR DESCRIPTION
**Pull request by jiaoew1991**
Signed-off-by: Enwei Jiao <enwei.jiao@zilliz.com>
issue: #20868

### CODE REVIEW

File: internal/util/etcd/etcd_util.go [hunk 1]
```diff
@@ -34,14 +33,21 @@ var (
 )
 
 // GetEtcdClient returns etcd client
-func GetEtcdClient(cfg *paramtable.EtcdConfig) (*clientv3.Client, error) {
```
Comment (jaime0815) [inline internal/util/etcd/etcd_util.go:37]: Why not use EtcdConfig as a parameter that will reduce a lot of redundant code?

Comment (jiaoew1991) [reply internal/util/etcd/etcd_util.go:37]: cannot do that, it will cause cycle dependency

File: internal/util/paramtable/base_table.go [hunk 2]
```diff
@@ -142,28 +143,35 @@ func (gp *BaseTable) initConfigsFromLocal(formatter func(key string) string) {
...
-		log.Info("cannot find etcd.endpoints")
+	etcdConfig := EtcdConfig{}
+	etcdConfig.Init(gp)
+	if etcdConfig.Endpoints.GetValue() == "" {
 		return
 	}
-	rootPath, err := gp.mgr.GetConfig("etcd.rootPath")
-	if err != nil {
-		log.Info("cannot find etcd.rootPath")
+	if etcdConfig.UseEmbedEtcd.GetAsBool() && !etcd.HasServer() {
```
Comment (congqixia) [inline internal/util/paramtable/base_table.go:151]: I think this line cause all the other places get messy. Maybe the dependency of etcd shall be removed here

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**